sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_repository.py

```diff
@@ -0,0 +1,228 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import TYPE_CHECKING
+
+from azure.core.exceptions import (
+    ClientAuthenticationError,
+    ResourceNotFoundError,
+    ResourceExistsError,
+    HttpResponseError,
+    map_error,
+)
+from azure.core.paging import ItemPaged
+from azure.core.pipeline import Pipeline
+from azure.core.tracing.decorator import distributed_trace
+
+from ._base_client import ContainerRegistryBaseClient, TransportWrapper
+from ._generated.models import AcrErrors
+from ._helpers import _parse_next_link
+from ._models import (
+    DeleteRepositoryResult,
+    ArtifactManifestProperties,
+    RepositoryProperties,
+)
+from ._registry_artifact import RegistryArtifact
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+    from azure.core.credentials import TokenCredential
+    from ._models import ContentProperties
+
+
+class ContainerRepository(ContainerRegistryBaseClient):
+    def __init__(self, endpoint, name, credential, **kwargs):
+        # type: (str, str, TokenCredential, Dict[str, Any]) -> None
+        """Create a ContainerRepository from an endpoint, repository name, and credential
+
+        :param str endpoint: An ACR endpoint
+        :param str name: The name of a repository
+        :param credential: The credential with which to authenticate
+        :type credential: :class:`~azure.core.credentials.TokenCredential`
+        :returns: None
+        :raises: None
+        """
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
+            endpoint = "https://" + endpoint
+        self._endpoint = endpoint
+        self.name = name
```

Comment:
Should this also have something like the property on `RegistryArtifact`:
```python
self.fully_qualified_name = self._endpoint + self.repository
```

Reply:
Yes adding it now.